src/mmrelay/plugin_loader.py

```
@@ -2229,13 +2603,22 @@ def load_plugins_from_directory(directory: str, recursive: bool = False) -> list
 
                         # Try to automatically install the missing dependency
                         try:
-                            if not _check_auto_install_enabled(config):
+                            if plugin_type == PLUGIN_TYPE_COMMUNITY:
                                 _raise_install_error(missing_pkg)
                             # Check if we're running in a pipx environment
                             in_pipx = (
                                 "PIPX_HOME" in os.environ
                                 or "PIPX_LOCAL_VENVS" in os.environ
                             )
+                            global _community_dep_install_warning_logged
```

Ruff (0.15.10)

[warning] Using the global statement to update `_community_dep_install_warning_logged` is discouraged

(PLW0603)